c2/c2-client-bundle/c2-client-service/src/main/java/org/apache/nifi/c2/client/service/operation/DebugOperationHandler.java

```diff
@@ -0,0 +1,257 @@
+        C2OperationState state = new C2OperationState();
+        state.setState(operationState);
+        state.setDetails(details);
+        return state;
+    }
+
+    private Stream<Path> debugBundlePaths(String baseDir, DebugBundleFile... debugBundleFiles) {
+        return Stream.of(debugBundleFiles)
+            .map(bundleFile -> Paths.get(baseDir, bundleFile.getFileName()))
+            .filter(Files::exists)
+            .filter(Files::isRegularFile);
+    }
+
+    private List<Path> filterContent(List<Path> bundleFilePaths) {
+        List<Path> contentFilteredFilePaths = new ArrayList<>();
+        for (Path path : bundleFilePaths) {
+            String fileName = path.getFileName().toString();
+            try {
+                Path tempDirectory = createTempDirectory(null);
+                Path tempFile = Paths.get(tempDirectory.toAbsolutePath().toString(), fileName);
+                write(tempFile, (Iterable<String>) lines(path).filter(contentFilter)::iterator);
+                contentFilteredFilePaths.add(tempFile);
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+        return contentFilteredFilePaths;
+    }
+
+    public Optional<byte[]> createDebugBundle(List<Path> filePaths) {
+        if (filePaths.size() == 0) {
+            return empty();
+        }
+        ByteArrayOutputStream byteOutputStream = null;
+        try {
+            byteOutputStream = new ByteArrayOutputStream();
+            try (GzipCompressorOutputStream gzipCompressorOutputStream = new GzipCompressorOutputStream(byteOutputStream);
+                 TarArchiveOutputStream tarOutputStream = new TarArchiveOutputStream(gzipCompressorOutputStream)) {
+                for (Path filePath : filePaths) {
+                    TarArchiveEntry tarArchiveEntry = new TarArchiveEntry(filePath.toFile(), filePath.getFileName().toString());
+                    tarOutputStream.putArchiveEntry(tarArchiveEntry);
+                    copy(filePath, tarOutputStream);
+                    tarOutputStream.closeArchiveEntry();
+                }
+                tarOutputStream.finish();
+            }
+        } catch (Exception e) {
+            LOG.error("Error during create compressed bundle", e);
+            closeQuietly(byteOutputStream);
+            byteOutputStream = null;
+        } finally {
+            closeQuietly(byteOutputStream);
+        }
+        return ofNullable(byteOutputStream).map(ByteArrayOutputStream::toByteArray);
+    }
+
+    private void cleanup(List<Path> paths) {
+        paths.forEach(path -> {
+            try {
+                deleteIfExists(path);
+            } catch (IOException e) {
+                LOG.warn("Unable to delete temporary file", e);
+            }
+        });
+    }
+
+    interface DebugBundleFile {
+        String getFileName();
+    }
+
+    enum LogFile implements DebugBundleFile {
+        APP_LOG_FILE("minifi-app.log"),
```

Comment:
The filenames depend on the logback configuration. It's not ideal to hardcode this. Can we use it from a property?

Reply:
Thanks for highlighting this. Now these are injected via properties